### PR TITLE
docs(website): [playground] add links to documentation and improve fixers

### LINKED ISSUE #5165: Docs: [playground] add link to rule documentation in list of fixers
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

linting result for eslint should have a link target="blank" that points to rule details

this can be achieved by reading rule url and should work for ts-eslint and eslint rules

### Affected URL(s)

https://typescript-eslint.io/play/

### LINKED ISSUE #5168: Docs: [playground] improve presentation of default fixer in right panel
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

if rule has only default fixer fixer should be displayed in same line instead of as `Fix this ** problem`

### Affected URL(s)

https://typescript-eslint.io/play/

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5168, fixes #5165
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

[Playground](https://deploy-preview-5169--typescript-eslint.netlify.app/play/#ts=4.7.2&sourceType=module&code=PTAEFpK6ZhYAUCUA7A9uApgWwA4BcBPcASxX0wCcAzAQwGNNFlZXXFEyKaHNQAhWpVCYAHhRQATAM6gAYmjSgA3qAC+HBJhQBXbKADK+Wvh2zliUKADyubaAC8oAIwAaS6ADCAGzTTM7ghqQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oFsbDbm+RNABmAQzKJ0UYdA7RIAGnDY8RUsgrU6DQa2LREI4fSaxuAppXyUxjAF7DUGSLPlKVAXxCegA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacGAC+IaUA)

- default fixer is no longer displaying redundant line and its positioned on same line as error,
- new `details` has been added that points to rule documentation provided by rule
  image
- rule name is correctly highlighted in tooltip and points to rule documentation
  image

### CODE REVIEW

File: packages/website/src/components/ErrorsViewer.tsx [hunk 1]
```diff
@@ -46,30 +36,45 @@ function ErrorBlock({
...
+                  onClick={(): void => {
+                    fixer.fix();
+                    setIsLocked(true);
+                  }}
+                >
+                  fix
+                </button>
+              </div>
+            ))}
+          </div>
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/ErrorsViewer.tsx:76]: Nit: maybe make a `<FixButton>` component for these, to DRY the two class name strings? Also that way you can pass it `item.fix` as a prop and not need the `!` above.

File: packages/website/src/components/ErrorsViewer.tsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
-import React, { useEffect, useMemo, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import type Monaco from 'monaco-editor';
-import type { ErrorItem } from './types';
+import clsx from 'clsx';
 
+import type { ErrorItem, ErrorGroup } from './types';
+import LinkIcon from '@site/src/icons/link.svg';
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/ErrorsViewer.tsx:6]: Hmm, does Docusaurus not export a way to use their svg used for external header links?

File: packages/website/src/components/ErrorsViewer.tsx [hunk 3]
```diff
@@ -91,11 +91,21 @@ export default function ErrorsViewer({
...
           <div className="margin-top--sm" key={group}>
-            <h4>{group}</h4>
-            {data.map((item, index) => (
+            <h4>
+              {group}
+              {uri && (
+                <>
+                  {' - '}
+                  <a href={uri} target="_blank">
+                    details <LinkIcon width={13.5} height={13.5} />
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/ErrorsViewer.tsx:103]: Nit: I personally associate "details" with per-instance details. Maybe that's just me.

```suggestion
                    docs <LinkIcon width={13.5} height={13.5} />
```

**Review by JoshuaKGoldberg — APPROVED** [review]
This is _fantastic_ 👏 Thanks!!

File: packages/website/src/components/ErrorsViewer.tsx (hunk 2 above)
Comment (Josh-Cena) [reply packages/website/src/components/ErrorsViewer.tsx:6]: Yes, there's `@theme/IconExternalLink`.

**Comment by armano2** [comment]
@JoshuaKGoldberg requested changes applied

**Review by JoshuaKGoldberg — APPROVED** [review]

**Pull request merged**